### PR TITLE
[Quant] Add initial Executorch BackendConfig

### PR DESCRIPTION
**Pull request by andrewor14**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85527

Summary: This commit adds the initial BackendConfig for backends
PyTorch lowers to through the Executorch stack. This initial
version is only intended to cover the following set of ops:

    quantized::linear_dynamic,
    quantized::add,
    quantized::batch_norm2d,
    quantized::conv2d.new,
    quantized::linear,
    quantized::conv2d_relu.new,
    aten::relu_,
    aten::_adaptive_avg_pool2d,
    aten::_reshape_alias_copy,
    aten::squeeze.dim,
    aten::permute

For now, the `BackendPatternConfig` for each of these ops is
the same as the ones for the corresponding ops in the FBGEMM
`BackendConfig`, though this may change in the future.

Reviewers: jerryzh168, vkuzo

Subscribers: jerryzh168, vkuzo

### CODE REVIEW

**Review by jerryzh168 — APPROVED** [review]
looks good, thanks!

File: torch/ao/quantization/backend_config/executorch.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+    Return the `BackendConfig` for backends PyTorch lowers to through the Executorch stack.
+    """
+    return BackendConfig("executorch") \
+        .set_backend_pattern_configs(_get_linear_configs()) \
+        .set_backend_pattern_configs(_get_conv_configs()) \
+        .set_backend_pattern_configs(_get_binary_ops_configs()) \
+        .set_backend_pattern_configs(_get_share_qparams_ops_configs()) \
+        .set_backend_pattern_configs(_get_bn_configs())
+
+__all__ = [
```
Comment (jerryzh168) [inline torch/ao/quantization/backend_config/executorch.py:214]: should this appear in the top of the file?

Comment (andrewor14) [reply torch/ao/quantization/backend_config/executorch.py:214]: Oh, yes. I just copied it over from FBGEMM. I'll move it
